docusaurus/docs/recipes/contentful.md

```diff
@@ -0,0 +1,511 @@
+
+Currently, Contentful doesn't enforce any limits on requests that hit their CDN cache. For requests that do hit the Content Delivery API enforces rate limits of 78 requests per second.
+
+When a client gets rate limited, the API responds with the `429 Too Many Requests` status code and sets the `X-Contentful-RateLimit-Reset` header that tells the client when it can make its next request. 
+
+### 2. Custom Caching Layer
+
+One preventive measure to avoid hitting the rate limit for Contentful is to implement your own custom caching layer.
+This can be done by setting up a proxy server which will add an `s-maxage` HTTP header into the Contentful's response.
+
+This header will then be interpreted by the _CDN_ that is delivering the application (for example _CloudFlare_), which will cache the response and avoid repeating the same request to Contentful during a specified time interval.
+
+The first thing you'll need in order to implement this solution is to install [http-proxy](https://www.npmjs.com/package/http-proxy):
+
+```sh
+$ npm i http-proxy
+```
+
+Then, you'll want to create an endpoint in your application which will serve as a proxy for all the requests directed at Contentful's API.
+
+For this, you'll have to create the file `pages/api/cms/[...cms].js`. This file/directory structure and naming is important because you'll want to receive any requests directed to `<hostname>/api/cms/*`.
+
+This endpoint will create the `proxy` server and, on each request, remove `/api/cms` from the request, rewrite the `host` header to the correct host (`cdn.contentful.com`), redirect the request to `cdn.contentful.com` and set the `Cache-Control` HTTP header of the response to `s-maxage=60`.
+
+```js
+import httpProxy from 'http-proxy';
+
+const proxy = httpProxy.createProxyServer({});
+
+export default (req, res) => {
+    req.url = req.url.replace('/api/cms', '');
+    req.headers.host = 'cdn.contentful.com';
+
+    proxy.web(req, res, {
+        target: {
+            host: 'cdn.contentful.com',
+        },
+    });
+
+    if (req.method === 'GET') {
+        res.setHeader('Cache-Control', 's-maxage=60');
+    }
+};
+```
+
+Finally, you will need to setup the Contentful client to direct it's requests to the endpoint you just configured. Please note that you only want to cache the requests when in a production environment.
+
+```js
+const clientOptions = {
+    space: process.env.CONTENTFUL_SPACE_ID,
+    accessToken: process.env.CONTENTFUL_TOKEN,
+};
+
+if (process.env.NODE_ENV === 'production') {
+    clientOptions.host = process.env.SITE_URL;
+    clientOptions.basePath = '/api/contentful';
+}
+
+const client = createClient(clientOptions);
+```
+
+**NOTES:**
+> ❗️ You have to be able to access to `process.env.NODE_ENV` from the client side as well as the server side, to proxy client side requests to Contentful in production environments.
```

Comment:
Nifty pick: always add new line after `:`, check other places too.

Moreover I would use the warning emoji if applicable to keep consistency with other recipes.